editor/editor_node.cpp

```diff
@@ -3455,82 +3455,169 @@ void EditorNode::set_addon_plugin_enabled(const String &p_addon, bool p_enabled,
 		addon_path = "res://addons/" + addon_path + "/plugin.cfg";
 	}
 
-	ERR_FAIL_COND(p_enabled && addon_name_to_plugin.has(addon_path));
-	ERR_FAIL_COND(!p_enabled && !addon_name_to_plugin.has(addon_path));
+	ERR_FAIL_COND(p_enabled && addon_name_to_plugins.has(addon_path));
+	ERR_FAIL_COND(!p_enabled && !addon_name_to_plugins.has(addon_path));
 
 	if (!p_enabled) {
-		EditorPlugin *addon = addon_name_to_plugin[addon_path];
-		remove_editor_plugin(addon, p_config_changed);
-		memdelete(addon);
-		addon_name_to_plugin.erase(addon_path);
+		// Disable
+		Vector<EditorPlugin *> plugins = addon_name_to_plugins[addon_path];
+		addon_name_to_plugins.erase(addon_path);
+		for (EditorPlugin *plugin : plugins) {
+			remove_editor_plugin(plugin, p_config_changed);
+			memdelete(plugin);
+		}
 		_update_addon_config();
 		return;
 	}
 
-	Ref<ConfigFile> cf;
-	cf.instantiate();
+	// Enable
+
 	if (!DirAccess::exists(addon_path.get_base_dir())) {
 		_remove_plugin_from_enabled(addon_path);
 		WARN_PRINT("Addon '" + addon_path + "' failed to load. No directory found. Removing from enabled plugins.");
 		return;
 	}
+
+	Ref<ConfigFile> cf;
+	cf.instantiate();
 	Error err = cf->load(addon_path);
 	if (err != OK) {
 		show_warning(vformat(TTR("Unable to enable addon plugin at: '%s' parsing of config failed."), addon_path));
 		return;
 	}
 
-	if (!cf->has_section_key("plugin", "script")) {
-		show_warning(vformat(TTR("Unable to find script field for addon plugin at: '%s'."), addon_path));
+	Vector<String> plugin_classes;
+	if (cf->has_section_key("plugin", "script")) {
+		// Backward compatibility
+		plugin_classes.push_back(cf->get_value("plugin", "script"));
+
+	} else if (cf->has_section_key("plugin", "editor_plugins")) {
+		Array array = cf->get_value("plugin", "script");
```

Comment:
That's a mistake. Should be `Array array = cf->get_value("plugin", "editor_plugins");`